tools/test-framework/src/util/proposal_status.rs

```diff
@@ -0,0 +1,79 @@
+use serde_json::Value;
+
+use crate::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ProposalStatus {
+    Unspecified = 0,
+    DepositPeriod = 1,
+    VotingPeriod = 2,
+    Passed = 3,
+    Rejected = 4,
+    Failed = 5,
+}
+
+impl TryFrom<i64> for ProposalStatus {
+    type Error = Error;
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::Unspecified),
+            1 => Ok(Self::DepositPeriod),
+            2 => Ok(Self::VotingPeriod),
+            3 => Ok(Self::Passed),
+            4 => Ok(Self::Rejected),
+            5 => Ok(Self::Failed),
+            _ => Err(Error::generic(eyre!(
+                "unknown value for proposal status: `{value}`"
+            ))),
+        }
+    }
+}
+
+impl TryFrom<String> for ProposalStatus {
```

Comment:
This is more idiomatic as a `std::str::FromStr` instance.

Reply:
Fixed in a8434f6612acd624fb558609bdeb242dc9da89bb